schema.yaml

```diff
@@ -0,0 +1,257 @@
+"$schema": https://json-schema.org/draft/2020-12/schema#
+title: CDBA Configuration Schema
+type: object
+properties:
+  devices:
+    type: array
+    items:
+      type: object
+      properties:
+        board:
+          type: string
+
+        name:
+          type: string
+
+        description:
+          type: string
+
+        console:
+          type: string
+          pattern: "^/dev/.*$"
+
+        fastboot:
+          type: string
+          pattern: "^[0-9a-f]{8}$"
+
+        fastboot_set_active:
+          type: boolean
+
+        broken_fastboot_boot:
+          type: boolean
+
+        usb_always_on:
+          type: boolean
+
+        fastboot_key_timeout:
+          type: integer
+          minimum: 1
+
+        cdba:
+          type: string
+          pattern: "^/dev/.*$"
+
+        conmux:
+          type: string
+          pattern: "^/dev/.*$"
+
+        external:
+          type: string
+
+        ppps_path:
+          type: string
+
+        ppps3_path:
+          type: string
+
+        qcomlt_debug_board:
+          type: string
+          pattern: "^/dev/.*$"
+
+        voltage:
+          type: integer
+
+        alpaca:
+          type: string
+          pattern: "^/dev/.*$"
+
+        users:
+          type: array
+          uniqueItems: true
+          minItems: 1
+          items:
+            type: string
+
+        ftdi_gpio:
+          oneOf:
+            - type: string
+              # <libftdi description>;[<interface>[;<gpios>...]]
+              # - libftdi description: ftdi_usb_open_string() formats
+              # - interface: A, B, C or D (default A)
+              # - gpios: type,id,polarity
+              #  - type: POWER, FASTBOOT_KEY, POWER_KEY or USB_DISCONNECT
+              #  - id: 0, 1, 2, 3, 4, 5, 6 or 7
+              #  - polarity: ACTIVE_HIGH or ACTIVE_LOW
+              pattern: "^(d:/.*)|(i:0x[0-9a-fA-F]{4}:0x[0-9a-fA-F]{4}(:[0-9]+)?)|(s:0x[0-9a-fA-F]{4}:0x[0-9a-fA-F]{4}:FT[0-9a-zA-Z]{6})(;[A-D](;(POWER|FASTBOOT_KEY|POWER_KEY|USB_DISCONNECT),[0-7],(ACTIVE_HIGH|ACTIVE_LOW)){0,4})?$"
+              deprecated: true
+
+            - type: object
+              properties:
+                description:
+                  type: string
+                  # libftdi ftdi_usb_open_string() formats
+                  # d:<devicenode>
+                  # i:<vendor>:<product>
+                  # i:<vendor>:<product>:<index>
+                  # s:<vendor>:<product>:<serial>
+                  pattern: "^(d:/.*)|(i:0x[0-9a-fA-F]{4}:0x[0-9a-fA-F]{4}(:[0-9]+)?)|(s:0x[0-9a-fA-F]{4}:0x[0-9a-fA-F]{4}:FT[0-9a-zA-Z]{6})$"
+                interface:
+                  enum:
+                    - A
+                    - B
+                    - C
+                    - D
+                gpios:
+                  type: array
+                  unevaluatedItems: false
+                  minItems: 1
+                  prefixItems:
+                    - type: object
```

Comment:
I think these can use a common definition.

Reply:
I planned to add common definitions at some point after figuring out the schema